### PR TITLE
turns radial menus into screen objects

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
turns radial menus into screen objects, this means you can now use them while in objects such as a closet, you also don't need to play silly games with your mouse movements while using say, a cult structure
This removes the radial menu opening animation as animate currently does not support it
Fixes some jank where a radial menu wouldn't work randomly
 
## Why It's Good For The Game
It's a bit of a pain not being able to use radial menus in objects, this makes them easier to use and faster to use

## Testing
clicked with radial menus

## Changelog
:cl:
tweak: Radial menus can now be used in objects
tweak: Removes the radial menu opening animation
/:cl:

### CODE REVIEW

File: code/_onclick/hud/radial.dm [hunk 1]
```diff
@@ -245,24 +249,15 @@ GLOBAL_LIST_EMPTY(radial_menus)
...
 	//Blank
-	menu_holder = image(icon = 'icons/effects/effects.dmi', loc = anchor, icon_state = "nothing", layer = ABOVE_HUD_LAYER)
-	menu_holder.appearance_flags |= KEEP_APART
-	menu_holder.vis_contents += elements + close_button
-	current_user.images += menu_holder
-
-/datum/radial_menu/proc/hide()
-	if(current_user)
-		current_user.images -= menu_holder
+	close_button.screen_loc = "CENTER:[pixel_x_difference], CENTER:[pixel_y_difference]"
```
Comment (Contrabang) [inline code/_onclick/hud/radial.dm:255]: Weirdly, for default by screenloc we don't have spaces after the commas
```suggestion
	close_button.screen_loc = "CENTER:[pixel_x_difference],CENTER:[pixel_y_difference]"
```

File: code/_onclick/hud/radial.dm [hunk 2]
```diff
@@ -165,20 +174,15 @@ GLOBAL_LIST_EMPTY(radial_menus)
...
-		starting.Scale(0.1, 0.1)
-		E.transform = starting
-		var/matrix/TM = matrix()
-		animate(E, pixel_x = px, pixel_y = py, transform = TM, time = timing)
-	else
-		E.pixel_y = py
-		E.pixel_x = px
+	E.pixel_y = py
+	E.pixel_x = px
+	E.screen_loc = "CENTER:[E.pixel_x + pixel_x_difference], CENTER:[E.pixel_y + pixel_y_difference]"
```
Comment (Contrabang) [inline code/_onclick/hud/radial.dm:183]:
```suggestion
	E.screen_loc = "CENTER:[E.pixel_x + pixel_x_difference],CENTER:[E.pixel_y + pixel_y_difference]"
```

File: code/_onclick/hud/radial.dm [hunk 3]
```diff
@@ -165,20 +174,15 @@ GLOBAL_LIST_EMPTY(radial_menus)
 	E.choice = null
 	E.next_page = FALSE
 
-/datum/radial_menu/proc/SetElement(obj/screen/radial/slice/E, choice_id, angle, anim, anim_order)
+/datum/radial_menu/proc/SetElement(obj/screen/radial/slice/E, choice_id, angle)
 	//Position
 	var/py = round(cos(angle) * radius) + py_shift
 	var/px = round(sin(angle) * radius)
```
Comment (Contrabang) [inline code/_onclick/hud/radial.dm:180]: Could you just merge these into the assignments of pixel_y and pixel_x below this?

File: code/_onclick/hud/radial.dm [hunk 4]
```diff
@@ -143,18 +150,20 @@ GLOBAL_LIST_EMPTY(radial_menus)
...
 	pages = page
 	current_page = 1
-	update_screen_objects(anim = entry_animation)
+	update_screen_objects()
 
-/datum/radial_menu/proc/update_screen_objects(anim = FALSE)
+/datum/radial_menu/proc/update_screen_objects()
 	var/list/page_choices = page_data[current_page]
 	var/angle_per_element = round(zone / page_choices.len)
+	pixel_x_difference = (world.icon_size * anchor.x + anchor.step_x + anchor.pixel_x) - (world.icon_size * current_user.mob.x + current_user.mob.step_x + current_user.mob.pixel_x)
```
Comment (Contrabang) [inline code/_onclick/hud/radial.dm:158]: Readability
```suggestion
	pixel_x_difference = ((world.icon_size * anchor.x) + anchor.step_x + anchor.pixel_x) - ((world.icon_size * current_user.mob.x) + current_user.mob.step_x + current_user.mob.pixel_x)
```

File: code/_onclick/hud/radial.dm [hunk 5]
```diff
@@ -143,18 +150,20 @@ GLOBAL_LIST_EMPTY(radial_menus)
...
+	pixel_y_difference = (world.icon_size * anchor.y + anchor.step_y + anchor.pixel_y) - (world.icon_size * current_user.mob.y + current_user.mob.step_y + current_user.mob.pixel_y)
```
Comment (Contrabang) [inline code/_onclick/hud/radial.dm:159]:
```suggestion
	pixel_y_difference = ((world.icon_size * anchor.y) + anchor.step_y + anchor.pixel_y) - ((world.icon_size * current_user.mob.y) + current_user.mob.step_y + current_user.mob.pixel_y)
```

File: code/_onclick/hud/radial.dm [hunk 6]
```diff
@@ -245,24 +249,15 @@ GLOBAL_LIST_EMPTY(radial_menus)
 		update_screen_objects()
 
 /datum/radial_menu/proc/show_to(mob/M)
-	if(current_user)
-		hide()
 	if(!M.client || !anchor)
 		return
-	current_user = M.client
 	//Blank
```
Comment (Contrabang) [inline code/_onclick/hud/radial.dm:254]: Is this comment needed?
```suggestion
```

File: code/_onclick/hud/radial.dm [hunk 7]
```diff
@@ -73,17 +81,16 @@ GLOBAL_LIST_EMPTY(radial_menus)
...
-	if(!istype(AM))
+	if(!istype(anchor))
 		return
+	current_user = user.client
 	var/mob/living/carbon/H
 	if(ishuman(user))
 		H = user
-	if((AM in user.client.screen) || (H && (AM in H.internal_organs)))
+	if((anchor in user.client.screen) || (H && (anchor in H.internal_organs)))
 		if(hudfix_method)
```
Comment (Contrabang) [inline code/_onclick/hud/radial.dm:94]: This seems to always be true. Is anything below this needed besides the anchor = user?

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by DGamerL — CHANGES_REQUESTED** [review]
Please adress Contra's reviews, this is going stale

**Review by S34NW — CHANGES_REQUESTED** [review]
Please do address these reviews within a few days or this will be closed. PR is extremely stale.

File: code/_onclick/hud/radial.dm (hunk 1 above)
Comment (GDNgit) [reply code/_onclick/hud/radial.dm]: Dupe code removed, reviews addressed

File: code/_onclick/hud/radial.dm (hunk 6 above)
Comment (GDNgit) [reply code/_onclick/hud/radial.dm]: maybe

**Review by Contrabang — COMMENTED** [review]
Tentative approval, this all looks good to me, but I'm not a master in screen objects

**Review by DGamerL — APPROVED** [review]
LGTM

**Review by Burzah — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Comment by GreatGatherer** [comment]
Breaks AI radial menu for HUDs

**Pull request merged**